Ghidra/Features/Decompiler/ghidra_scripts/classrecovery/RecoveredClassHelper.java

```diff
@@ -7826,25 +7828,9 @@ private FunctionDefinition updateFunctionDefinition(FunctionDefinition functionD
 		ParameterDefinition[] currentArgs = functionDefinition.getArguments();
 		ParameterDefinition[] changedArgs = newFunctionDefinition.getArguments();
 
-		// only update if there are differences and the func sigs have same length
-		// if they don't have same length then something was possibly overridden in a child and
-		// it needs to stay the same as it was except the name
-		if (!currentArgs.equals(changedArgs) && (currentArgs.length == changedArgs.length)) {
-			ParameterDefinition[] newArgs = new ParameterDefinition[currentArgs.length];
-			for (int i = 0; i < currentArgs.length; i++) {
-				if (currentArgs[i].getName().equals("this") ||
-					currentArgs[i].equals(changedArgs[i])) {
-					newArgs[i] = currentArgs[i];
-				}
-				else {
-					newArgs[i] = changedArgs[i];
-					changed = true;
-				}
-			}
-			if (changed) {
-				functionDefinition.setArguments(newArgs);
-			}
-
+		// only update if there are differences
+		if (!currentArgs.equals(changedArgs)) {
```

Comment:
The equals does not work correctly. For example, comparing two empty arrays returns not equals. Per our data type expert, in general, using equals is not recommended when comparing data types. Will accept rest of PR but not this part. Thank you for the submission!

Reply:
I am reworking that area of code in the PR branch I created off your original PR changes. I think once I commit the changes and they get pushed out this ticket will be auto-updated to let you know.